blerpc/src/main/java/com/blerpc/AnnotationMessageConverter.java

```diff
@@ -0,0 +1,309 @@
+                    break;
+                case LONG:
+                    messageBuilder.setField(fieldDescriptor, deserializeLong(value, relativeFieldBytesRange, fieldName));
+                    break;
+                case ENUM:
+                    messageBuilder.setField(fieldDescriptor, fieldDescriptor.getEnumType()
+                            .findValueByNumber((int) deserializeLong(value, relativeFieldBytesRange, fieldName)));
+                    break;
+                case BOOLEAN:
+                    messageBuilder.setField(fieldDescriptor, deserializeBoolean(value, relativeFieldBytesRange));
+                    break;
+                // TODO(#5): Add support of ByteString, String, Float and Double.
+                default:
+                    checkArgument(false, "Unsupported field type: %s, field name: %s", fieldType.name(), fieldName);
+            }
+        }
+        return messageBuilder.build();
+    }
+
+    private long deserializeLong(byte[] bytes, BytesRange bytesRange, String fieldName) {
+        int firstByte = bytesRange.getFromByte();
+        int lastByte = bytesRange.getToByte();
+        int bytesSize = lastByte - firstByte;
+        checkArgument(bytesSize <= 8,
+                "Only integer fields with declared bytes size in [1, 8] are supported. Field %s has %s bytes size.",
+                fieldName,
+                bytesSize);
+
+        long result = 0;
+        if (byteOrder.equals(ByteOrder.BIG_ENDIAN)) {
+            for (int i = firstByte; i < lastByte; i++) {
+                result <<= 8;
+                result |= bytes[i] & 0xFF;
+            }
+        } else {
+            for (int i = lastByte; i > firstByte; i--) {
+                result <<= 8;
+                result |= bytes[i - 1] & 0xFF;
+            }
+        }
+        return result;
+    }
+
+    private boolean deserializeBoolean(byte[] bytes, BytesRange bytesRange) {
+        return bytes[bytesRange.getFromByte()] != 0;
+    }
+
+    private static int getMessageBytesSize(Message message) {
+        Descriptor descriptor = message.getDescriptorForType();
+        MessageOptions options = descriptor.getOptions();
+        checkArgument(options.hasExtension(Blerpc.messageSize) || descriptor.getFields().isEmpty(),
+                "Proto message %s with fields must have BytesSize option",
+                descriptor.getName());
+        return options.getExtension(Blerpc.messageSize).getMessageSizeBytes();
+    }
+
+    private static void checkHasExpectedBytesSize(Message message, BytesRange bytesRange) {
+        int messageBytesSize = getMessageBytesSize(message);
+        int expectedBytesSize = bytesRange.getToByte() - bytesRange.getFromByte();
+        checkArgument(messageBytesSize == expectedBytesSize,
+                "Field bytes size %s is not equals to message %s bytes size %s related to this field",
```

Comment:
"Non-primitive field %s has declared size %d, which is not equal to the size of it's type %d."